### PR TITLE
Start listening to more Events. Log more location information.

### PR DESCRIPTION
**Pull request by knowuh**
* Removed useless helper class "context" and made that all part of the logger itself.
* Added more serialized fields for logging context (lastPosition, lastHeading, selectedStar) &etc.
* Light wrapper around events

[#170563535] Add More logging

https://www.pivotaltracker.com/story/show/170563535

### CODE REVIEW

File: Assets/Scripts/Utilities/CCLogger.cs [hunk 1]
```diff
@@ -41,20 +48,191 @@ private static string getSceneName()
...
+        }
+        else
+        {
+            selectedObject = "";
+        }
+
+    }
+
+
+    private void RegisterEventListeners()
```
Comment (dstrawberrygirl) [inline Assets/Scripts/Utilities/CCLogger.cs]: We will need to remove listeners at the end of running the application - Unity is really sticky when it comes to listeners.

File: Assets/Scripts/Utilities/CCLogger.cs [hunk 2]
```diff
@@ -41,20 +48,191 @@ private static string getSceneName()
...
+
+    private static double getTimeMillies()
+    {
+        TimeSpan t = DateTime.UtcNow - new DateTime(1970, 1, 1);
+        return t.TotalMilliseconds;
+    }
+
+    private static string getSceneName()
+    {
+        UnityEngine.SceneManagement.Scene scene = UnityEngine.SceneManagement.SceneManager.GetActiveScene();
```
Comment (dstrawberrygirl) [inline Assets/Scripts/Utilities/CCLogger.cs]: can clean this up a bit if you add a Using statement up top to UnityEngine.SceneManagement then just return SceneManager.GetActiveScene().name;

File: Assets/Scripts/Utilities/CCLogger.cs [hunk 3]
```diff
@@ -78,25 +256,31 @@ private async Task postLog(string jsonLogMessage)
...
-        await postLog(json);
+        @event = eventType;
+        event_value = eventType;
+        message = msg;
+        UpdateContext();
+        bool devMode = SimulationManager.GetInstance().server == ServerList.Local;
+        string json = JsonUtility.ToJson(this, true);
+
+        // TBD: This is really helpful right now, but might get noisy.
+        Debug.Log(json);
```
Comment (dstrawberrygirl) [inline Assets/Scripts/Utilities/CCLogger.cs]: Could wrap this in conditional if this is just for use in-editor
```
#if UNITY_EDITOR
Debug.Log(json);
#endif
```

**Comment by knowuh** [comment]
OK, I think we are ready for another review.

**Pull request merged**